ocaml/libs/tracing/tracing.ml

```diff
@@ -466,6 +444,14 @@ module Export = struct
 
   let set_export_interval t = export_interval := t
 
+  let host_id = ref "localhost"
+
+  let set_host_id id = host_id := id
+
+  let service_name = ref "unknown"
+
+  let set_service_name name = service_name := name
```

Comment:
I would think having unkowns is a rather unwanted effect, would it be worth using a function to log when an unknown is generated?

```suggestion
  let service_name = ref None
  let set_service_name name = service_name := Some name
  let get_service_name () =
    match !sercive_name with
    | None ->
        warn "service name not yet set!" ;
        "unknown"
    | Some name ->
        name
```